python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/antlr_asset_selection.py

```diff
@@ -1,24 +1,154 @@
 from antlr4 import CommonTokenStream, InputStream
+from antlr4.error.ErrorListener import ErrorListener
 
-from dagster._annotations import experimental
 from dagster._core.definitions.antlr_asset_selection.generated.AssetSelectionLexer import (
     AssetSelectionLexer,
 )
 from dagster._core.definitions.antlr_asset_selection.generated.AssetSelectionParser import (
     AssetSelectionParser,
 )
+from dagster._core.definitions.antlr_asset_selection.generated.AssetSelectionVisitor import (
+    AssetSelectionVisitor,
+)
+from dagster._core.definitions.asset_selection import AssetSelection, CodeLocationAssetSelection
+from dagster._core.storage.tags import KIND_PREFIX
+
+
+class AntlrInputErrorListener(ErrorListener):
+    def syntaxError(self, recognizer, offendingSymbol, line, column, msg, e):
+        raise Exception(f"Syntax error at line {line}, column {column}: {msg}")
+
+
+class AntlrAssetSelectionVisitor(AssetSelectionVisitor):
+    def visitStart(self, ctx: AssetSelectionParser.StartContext):
+        return self.visit(ctx.expr())
+
+    def visitAssetExpression(self, ctx: AssetSelectionParser.AssetExpressionContext):
+        return self.visit(ctx.assetExpr())
+
+    def visitParenthesizedExpression(
+        self, ctx: AssetSelectionParser.ParenthesizedExpressionContext
+    ):
+        return self.visit(ctx.expr())
+
+    def visitUpTraversalExpression(self, ctx: AssetSelectionParser.UpTraversalExpressionContext):
+        selection: AssetSelection = self.visit(ctx.expr())
+        traversal_depth = self.visit(ctx.traversal())
+        return selection.upstream(depth=traversal_depth)
+
+    def visitAndExpression(self, ctx: AssetSelectionParser.AndExpressionContext):
+        left: AssetSelection = self.visit(ctx.expr(0))
+        right: AssetSelection = self.visit(ctx.expr(1))
+        return left & right
+
+    def visitNotExpression(self, ctx: AssetSelectionParser.NotExpressionContext):
+        selection: AssetSelection = self.visit(ctx.expr())
+        return AssetSelection.all() - selection
```

Comment:
Should it be `AssetSelection.all(include_sources=True)` here?

Reply:
Hmm, good question - I would say yes, though this does raise the q of if we need syntax for `MaterializableAssetSelection` https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/definitions/asset_selection.py#L801

Reply:
@benpankow we could do something like `materializable:true`